File: fuentes/sapiens/src/main/java/com/vortexbird/sapiens/service/UsuarioService.java
package com.vortexbird.sapiens.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.vortexbird.sapiens.domain.Usuario;
import com.vortexbird.sapiens.dto.CargueMasivoDTO;
import com.vortexbird.sapiens.dto.UsuarioDTO;




/**
* @author Zathura Code Generator http://zathuracode.org/
* www.zathuracode.org
*
*/
public interface UsuarioService extends GenericService<Usuario, Integer> {

	UsuarioDTO login(String codigo, String password) throws Exception;

	public Page<UsuarioDTO> getUsuariosPorTipo(Integer tiusId, Integer semestre, Integer progId, Integer facuId, Integer grupId,String filtro, int pageNumber, int pageSize) throws Exception;
	
	public List<Integer> getAllUsuariosPorTipo(Integer tiusId, Integer semestre, Integer progId, Integer facuId, Integer grupId,String filtro) throws Exception;

	String getNombreUsuario(Integer usuaId) throws Exception;

	void guardar(UsuarioDTO usuarioDTO) throws Exception;

	String solicitarClave(String codigo) throws Exception;

	void recuperarClave(String token) throws Exception;
	
	void cambiarClave(UsuarioDTO usuarioDTO) throws Exception;

	List<String> cargar(CargueMasivoDTO request) throws Exception;

	void recrearClaves(Long usuarioCreador) throws Exception;

	List<UsuarioDTO> getUsuariosPorGrupo(Integer grupId) throws Exception;
	
	String getCorreoUsuarioPorCodigo(String codigo) throws Exception;
}

File: fuentes/sapiens-frontend/src/app/services/usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { CargueMasivo } from 'app/domain/cargue-masivo';
import { Usuario } from 'app/domain/usuario';
import { global } from 'app/utils/global';
import { environment } from 'environments/environment';
import { Observable } from 'rxjs';
import { LocalStorageService } from './local-storage.service';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {

  url: string;

  constructor(private httpClient: HttpClient, private localStorage: LocalStorageService) {
    this.url = environment.URL + 'usuario/';
  }

  public getUsuario(): Usuario {
    let usuario: Usuario = this.localStorage.getFromLocal(global.SESSION_ITEMS.USUARIO);

    return usuario;
  }

  public findById(usuaId: number): Observable<any> {
    return this.httpClient.get(this.url + 'findById/' + usuaId);
  }

  public login(usuario: Usuario): Observable<any> {
    return this.httpClient.post(this.url + 'login/', usuario);
  }

  public guardar(usuario: Usuario): Observable<any> {
    return this.httpClient.post(this.url + 'guardar/', usuario);
  }

  public getUsuariosPorTipo(tiusId: number, semestre: number, progId: number, facuId:number, grupId:number, filtro: string, pageNumber: number, pageSize: number): Observable<any> {
    if (!filtro || filtro.length == 0 || filtro === '%') {
      filtro = "*";
    }
    if (!facuId || facuId <=0 || facuId.toString() == "") {
      facuId = -1;
    }
    if (!semestre || semestre <=0 || semestre.toString() == "") {
      semestre = -1;
    }
    if (!progId || progId <=0 || progId.toString() == "") {
      progId = -1;
    }
    if (!grupId || grupId <=0 || grupId.toString() == "") {
      grupId = -1;
    }
    return this.httpClient.get(this.url + 'getUsuariosPorTipo/' + tiusId + "/" + semestre + "/" + progId + "/" + facuId + "/" + grupId + "/" + filtro + "/" + pageNumber + "/" + pageSize);
  }

  public getAllUsuariosPorTipo(tiusId: number, semestre: number, progId: number, facuId:number, grupId:number, filtro: string): Observable<any> {
    if (!filtro || filtro.length == 0 || filtro === '%') {
      filtro = "*";
    }
    if (!facuId || facuId <=0 || facuId.toString() == "") {
      facuId = -1;
    }
    if (!semestre || semestre <=0 || semestre.toString() == "") {
      semestre = -1;
    }
    if (!progId || progId <=0 || progId.toString() == "") {
      progId = -1;
    }
    if (!grupId || grupId <=0 || grupId.toString() == "") {
      grupId = -1;
    }
    return this.httpClient.get(this.url + 'getAllUsuariosPorTipo/' + tiusId + "/" + semestre + "/" + progId + "/" + facuId + "/" + grupId + "/" + filtro);
  }

  public solicitarClave(codigo: string): Observable<any> {
    return this.httpClient.get(this.url + 'solicitarClave/' + codigo);
  }

  public cambiarClave(request: Usuario): Observable<any> {
    return this.httpClient.post(this.url + 'cambiarClave/', request);
  }

  public cargar(request: CargueMasivo): Observable<any> {
    return this.httpClient.post(this.url + 'cargar/', request);
  }

  public getUsuariosPorGrupo(grupId: number): Observable<any> {
    return this.httpClient.post(this.url + 'getUsuariosPorGrupo/', grupId);
  }

  public getCorreoUsuarioPorCodigo(codigo: string): Observable<any> {
    return this.httpClient.post(this.url + 'getCorreoUsuarioPorCodigo/', codigo);
  }
}
